### DISCUSSION TITLE
Windows Package Manager v1.1.12653

### DISCUSSION THREAD

**Discussion by denelon** [microsoft/winget-cli#1515] [Releases]
This release represents our Windows Package Manager 1.1 release candidate build for Windows 10 (1809+). 

Experimental features have been disabled in this release. We will follow this release with another Pre-release "developer" build at GitHub so users can continue with experimental features available.

# Bugs
#797 Silent install of "winget install git.git" is not working
#1497 Make rename retry more frequently for longer, then try making a hardlink

<hr /><em>This discussion was created from the release <a href='https://github.com/microsoft/winget-cli/releases/tag/v1.1.12653'>Windows Package Manager v1.1.12653</a>.</em>

**Comment by jedieaston** [comment]
Can schema 1.1 be used in this version?

**Reply by denelon** [reply]
Yes, but much of the client implementation work still needs to be implemented.

**Comment by jedieaston** [comment]
Also, interestingly, searches seem to be a lot less strict than they used to be in this build (although this might be changes to the msstore source rather than client side changes):

```
PS C:\Users\easton> winget install --interactive Microsoft.VisualStudio.2019.Community
Multiple packages found matching input criteria. Please refine the input.
Name                         Id                                    Source
--------------------------------------------------------------------------
Victory Community Bank       9WZDNCRDC5H0                          msstore
Visual Studio Community 2019 Microsoft.VisualStudio.2019.Community winget
PS C:\Users\easton> winget --info
Windows Package Manager v1.1.12653
Copyright (c) Microsoft Corporation. All rights reserved.

Windows: Windows.Desktop v10.0.19043.928
Package: Microsoft.DesktopAppInstaller v1.16.12653.0

Logs: %LOCALAPPDATA%\Packages\Microsoft.DesktopAppInstaller_8wekyb3d8bbwe\LocalState\DiagOutputDir

Links
---------------------------------------------------------------------------
Privacy Statement   https://aka.ms/winget-privacy
License Agreement   https://aka.ms/winget-license
Third Party Notices https://aka.ms/winget-3rdPartyNotice
Homepage            https://aka.ms/winget
Windows Store Terms https://www.microsoft.com/en-us/storedocs/terms-of-sale
PS C:\Users\easton>
```

"Victory Community Bank" seems to be a good response for "Microsoft.VisualStudio.2019.Community".

**Reply by denelon** [reply]
You are correct (this is based on differences with the Microsoft Store source). The search behavior in the Microsoft Store was originally designed for a user browsing a visual gallery with product images and categories. We will continue to evaluate and refine the results over time. I am working on a blog post to explain some of the nuances of working with multiple sources.

**Comment by denelon** [comment]
# Some help with sources:

The default sources are "**winget**" for the Windows Package Manager Community app repository and "**msstore**" for the Microsoft Store.

Most commands interacting with sources (search, show, install) honor the `--source` or `-s` with the source specified after.
`winget search "Visual Studio Code" -s msstore`
`winget search vscode -s winget`

The store source does not have a moniker, so it's best to use the "Id" for installing packages from the store source.
```
PS C:\Users\denelon> winget search "visual studio"
Name                                         Id                                            Version             Source
-----------------------------------------------------------------------------------------------------------------------
Visual Studio Code                           XP9KHM4BK9FZ7Q                                Unknown             msstore
```
`winget install XP9KHM4BK9FZ7Q`

You may remove a source with `winget source remove <source> --force`, and you may reset them with `winget source reset` using an elevated prompt.

List and export don't have strong correlations with the store source yet, so they are not using the store source in this release.

**Comment by spragginsdesigns** [comment]
Can someone please post settings for winget in? JSON I can copy? 
I feel like after I installed these two Samsung EVO Pro 980 M.2 NVME SSDs, I messed up my. JSON schema and I want all the correct settings again if Could someone give me there's? @denelon, any chance you could send me your settings.JSON to austin@spragginsdesigns.com or through my web app https://www.spragginsdesigns.com, I am OSCP certified, and this is the safest way to send someone something on a website I am aware of. 

Mine looks terrible and nothing like what it was: 
`{
  "$id": "https://aka.ms/winget-settings.schema.json",
  "$schema": "https://json-schema.org/draft/2019-09/schema#",
  "title": "Microsoft's Windows Package Manager Settings Profile Schema",
  "definitions": {
    "Source": {
      "description": "Source repository settings",
      "type": "object",
      "properties": {
        "autoUpdateIntervalInMinutes": {
          "description": "Number of minutes before source update",
          "type": "integer",
          "default": 5,
          "minimum": 0,
          "maximum": 43200
        }
      }
    },
    "Visual": {
      "description": "Visual settings",
      "type": "object",
      "properties": {
        "progressBar": {
          "description": "Progress bar display style",
          "type": "string",
          "enum": ["accent", "rainbow", "plain", "retro"]
        }
      }
    },
    "InstallPrefReq": {
      "description": "Shared schema for preferences and requirements",
      "type": "object",
      "properties": {
        "scope": {
          "description": "The scope of a package install",
          "type": "string",
          "enum": ["user", "machine"],
          "default": "user"
        },
        "locale": {
          "description": "The locales of a package install",
          "type": "array",
          "items": {
            "type": "string",
            "pattern": "^([a-zA-Z]{2}|[iI]-[a-zA-Z]+|[xX]-[a-zA-Z]{1,8})(-[a-zA-Z]{1,8})*$",
            "maxLength": 20
          },
          "minItems": 1,
          "maxItems": 10
        }
      }
    },
    "InstallBehavior": {
      "description": "Install settings",
      "type": "object",
      "properties": {
        "preferences": {
          "$ref": "#/definitions/InstallPrefReq"
        },
        "requirements": {
          "$ref": "#/definitions/InstallPrefReq"
        }
      }
    },
    "Telemetry": {
      "description": "Telemetry settings",
      "type": "object",
      "properties": {
        "disable": {
          "description": "Controls whether telemetry events are written",
          "type": "boolean",
          "default": false
        }
      }
    },
    "Network": {
      "description": "Network settings",
      "type": "object",
      "properties": {
        "downloader": {
          "description": "Control which download code is used for packages",
          "type": "string",
          "enum": ["default", "wininet", "do"],
          "default": "default"
        },
        "doProgressTimeoutInSeconds": {
          "description": "Number of seconds to wait without progress before fallback",
          "type": "integer",
          "default": 60,
          "minimum": 1,
          "maximum": 600
        }
      }
    },
    "Experimental": {
      "description": "Experimental Features",
      "type": "object",
      "properties": {
        "experimentalCMD": {
          "description": "Reference implementation for an experimental command",
          "type": "boolean",
          "default": true
        },
        "experimentalARG": {
          "description": "Reference implementation for an experimental argument",
          "type": "boolean",
          "default": false
        },
        "dependencies": {
          "description": "Support for package dependencies",
          "type": "boolean",
          "default": false
        },
        "directMSI": {
          "description": "Enable use of MSI APIs rather than msiexec for MSI installs",
          "type": "boolean",
          "default": true
        }
      }
    }
  },
  "allOf": [
    {
      "properties": {
        "visual": {
          "$ref": "#/definitions/Visual"
        }
      },
      "additionalItems": true
    },
    {
      "properties": {
        "source": {
          "$ref": "#/definitions/Source"
        }
      },
      "additionalItems": true
    },
    {
      "properties": {
        "installBehavior": {
          "$ref": "#/definitions/InstallBehavior"
        }
      },
      "additionalItems": true
    },
    {
      "properties": {
        "telemetry": {
          "$ref": "#/definitions/Telemetry"
        }
      },
      "additionalItems": true
    },
    {
      "properties": {
        "network": {
          "$ref": "#/definitions/Network"
        }
      },
      "additionalItems": true
    },
    {
      "properties": {
        "experimentalFeatures": {
          "$ref": "#/definitions/Experimental"
        }
      },
      "additionalItems": true
    }
  ],
  "additionalProperties": true
}
`
Nothing on mine seems to work anymore, I have to go and get the installer now or use PS Gallery Package Management, yikes. 
![image](https://user-images.githubusercontent.com/43624199/135954708-74656e55-8689-4fe2-aca8-3b3c9361ca73.png)

**Comment by denelon** [comment]
```
{
    "$schema": "https://aka.ms/winget-settings.schema.json",

    // For documentation on these settings, see: https://aka.ms/winget-settings
    // "source": {
    //    "autoUpdateIntervalInMinutes": 5
    // },

    "visual": {"progressBar": "rainbow"},

}
```

**Reply by spragginsdesigns** [reply]
Thanks, this is what I got now and it's working better. 

```
       {
    
    "$schema": "https://aka.ms/winget-settings.schema.json",
    
    // For documentation on these settings, see: https://aka.ms/winget-settings
    
    // "source": {
    
    // "autoUpdateIntervalInMinutes": 5
    
    // },
    
    "visual": {
    
    "progressBar": "accent"
    
    },
    
    "source": {
    
    "autoUpdateIntervalInMinutes": 3
    
    },
    
    "telemetry": {
    
    "disable": true
    
    },
    
    "experimentalFeatures": {
    
    "experimentalCmd": true,
    
    "experimentalArg": true,
    
    "dependencies": true,
    
    "directMSI": true,
    
    "install": true,
    
    "remove": true,
    
    "experimentalMSStore": true
    
    }
    
    }
```

**Comment by nbevans** [comment]
From what I can tell it looks like this build of DesktopAppInstaller has been published on the Microsoft Store and is being rolled out to general public - can you confirm @denelon ?

**Reply by jedieaston** [reply]
I'm not @denelon, but it is. If you update App Installer in the Microsoft Store you get this build.

**Reply by denelon** [reply]
Yes, this build has been rolled out to Windows 10 (1809+).

**Comment by spragginsdesigns** [comment]
What is the difference between winget v1.1.12702, and DesktopAppInstaller? Not to mention App Installer, would love for someone to elaborate more on all that it takes to run the winget cli.

**Reply by denelon** [reply]
The Windows Package Manager is shipped inside the App Installer. If you run `winget --info` it will display the versions of both.

**Comment by Masamune3210** [comment]
DesktopAppInstaller = App Installer iirc, and winget is included in App Installer

**Reply by spragginsdesigns** [reply]
Thank you. I used to do `winget install dashlane` or `winget install chrome`, and it would work. It's been a few months of doing `winget install --id Dashlane.Dashlane` or whatever I am installing because it does not work the way it used to for me. 

![image](https://user-images.githubusercontent.com/43624199/138817543-27266e1d-b57c-46aa-a9b8-caa73eab3079.png)

**Comment by UseYourGadgets1** [comment]
Please add download size in the "Winget show _id_" command.

**Reply by jedieaston** [reply]
There's an issue for this: #1070.
